src/test/mock_ethers_provider.ts

```diff
@@ -160,6 +167,23 @@ export default class MockEthersProvider {
     when(this.getNetwork).calledWith().mockReturnValue({ chainId, ensAddress, name });
   }
 
+  public setTransaction(transaction: MockTransactionData) {
+    this.setTransactionResponse(transaction);
+    this.setTransactionReceipt(transaction);
+  }
+
+  public setTransactionResponse(transaction: MockTransactionData) {
+    when(this.getTransaction)
+      .calledWith(transaction.hash)
+      .mockReturnValue(Promise.resolve(transaction.getTransactionResponse()));
+  }
+
+  public setTransactionReceipt(transaction: MockTransactionData) {
+    when(this.getTransactionReceipt)
+      .calledWith(transaction.transactionHash)
+      .mockReturnValue(Promise.resolve(transaction.getTransactionReceipt()));
+  }
```

Comment:
I'm not sure in which cases we would need to set them separately. because we already set them like `mockTransactionData.setTransactionResponse()` and `mockTransactionData.setTransactionReceipt()`

I'm thinking if we should make these `private`?  🤔 wdyt?

Reply:
Ohh okay, I think we can do that. 
But let me know whether I have understood what you meant properly. 
Since we already have these in `MockTransactionData` and there is a `MockEthersprovider.setTransaction(transaction: MockTransactionData)` method that sets both `transactionResponse` and `transactionReceipt`, we can make these individual set methods in `MockEthersProvider` optional?

Reply:
yes, that's what mean 👍